mathext/hypergeo.go

```diff
@@ -0,0 +1,204 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/mathext/internal/cephes"
+)
+
+// The hypergeometric relations used here are from:
+//
+// M. Abramowitz and I. A. Stegun 1965. Handbook of Mathematical Functions, New York: Dover
+//
+// It is available online via https://personal.math.ubc.ca/~cbm/aands/frameindex.htm
+
+// Hypergeo returns the value of the Gaussian Hypergeometric function at z.
+// For |z| < 1, this implementation follows the Cephes library.
+// For |z| ≥ 1, this implementation performs analytic continuation via relevant Hypergeometric identities.
+//
+// See https://en.wikipedia.org/wiki/Hypergeometric_function for more details.
+func Hypergeo(a, b, c, z float64) float64 {
+	// Fix numerical issues for large |a|, |b| or |c| using equations 15.2.10 and 15.2.12.
+	// Simplify discussion by ensuring |a| > |b|.
+	aa, ab, ac := math.Abs(a), math.Abs(b), math.Abs(c)
+	if ab > aa {
+		return Hypergeo(b, a, c, z)
+	}
+	acFar := (aa/ac < 0.9 || ac/aa < 0.9)
+	if a < -50 && ac > 2 && acFar {
+		return eqn15_2_10(a, b, c, z)
+	}
+	if c < -50 && aa > 2 && acFar {
+		return eqn15_2_12(a, b, c, z)
+	}
+
+	// Canonical case |z| < 1 supported by Cephes.
+	if math.Abs(z) < 1 {
+		return cephes.Hyp2f1(a, b, c, z)
+	}
+
+	// Function undefined between the 1 and Inf branch points.
+	if z >= 1 {
+		return math.NaN()
+	}
+
+	// Special case for z=-1, use equation 15.3.4 in Abramowitz.
+	if z == -1 {
+		return eqn15_3_4(a, b, c, z)
+	}
+
+	// When a or b is a non-positive integer, Hypergeo reduces to the finite polynomial described in equation 15.4.1.
```

Comment:
For these internal comments, can you wrap them earlier; somewhere around 60-70 would be good. Also in the test files.

Reply:
Right, I have wrapped comments here and in the test files.
When deciding where to wrap, aside from around 60-70, I also tried to wrap when there's a comma.